From f2ffdce42a1f0d0fcb6ac4bdc887b7bc8f5586fd Mon Sep 17 00:00:00 2001
From: Peleg Vigodny <pvigodny@habana.ai>
Date: Wed, 14 Sep 2022 13:52:47 +0300
Subject: [PATCH] [DEVELOPERS] install checkpatch pre-commit hooks

Avoid unnecessary Jenkins fails for checkpatch failures
Cherry pick this change and ./install-pre-commit-hooks.sh
This will run the relevant checkpatch on your changes before
committing them

Change-Id: I89248f51b559a1ac52eb9d990b1573852e05d9bb
Signed-off-by: Peleg Vigodny <pvigodny@habana.ai>
---

diff --git a/install-pre-commit-hooks.sh b/install-pre-commit-hooks.sh
new file mode 100755
index 0000000..0fb356e
--- /dev/null
+++ b/install-pre-commit-hooks.sh
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+for dir in                                      \
+        linux                                   \
+        preboot; do
+        cp linux-pre-commit ../${dir}/.git/hooks/pre-commit
+done
+
+for dir in                                      \
+        zephyr                                  \
+        zephyr-plugins                          \
+        zephyr-apps/arcpid                      \
+        zephyr-apps/preboot                     \
+        zephyr-apps/arcmgmt                     \
+        zephyr-modules/boot_info                \
+        zephyr-modules/eeprom                   \
+        zephyr-modules/i2c-indirect; do
+        cp zephyr-pre-commit ../${dir}/.git/hooks/pre-commit
+done
diff --git a/linux-pre-commit b/linux-pre-commit
new file mode 100755
index 0000000..6899706
--- /dev/null
+++ b/linux-pre-commit
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+exec git diff --cached | ${HABANA_SOFTWARE_STACK}/embedded/linux/scripts/checkpatch.pl --no-signoff --ignore=GERRIT_CHANGE_ID,FILE_PATH_CHANGES,CAMELCASE  || true
diff --git a/zephyr-pre-commit b/zephyr-pre-commit
new file mode 100755
index 0000000..901cc6d
--- /dev/null
+++ b/zephyr-pre-commit
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+exec git diff --cached | ${HABANA_SOFTWARE_STACK}/embedded/zephyr/scripts/checkpatch.pl --no-signoff --ignore=GERRIT_CHANGE_ID,FILE_PATH_CHANGES,CAMELCASE,TRAILING_SEMICOLON,PREFER_KERNEL_TYPES,PRINTK_WITHOUT_KERN_LEVEL || true
